src/azure-cli/azure/cli/command_modules/appconfig/_kv_import_helpers.py

```diff
@@ -715,6 +743,160 @@ def __read_kv_from_app_service(
         raise CLIError("Failed to read key-values from appservice.\n" + str(exception))
 
 
+def __read_kv_from_kubernetes_configmap(
+    cmd,
+    aks_cluster,
+    configmap_name,
+    format_,
+    namespace="default",
+    prefix_to_add="",
+    content_type=None,
+    depth=None,
+    separator=None
+):
+    """
+    Read key-value pairs from a Kubernetes ConfigMap using aks_runcommand.
+
+    Args:
+        cmd: The command context object
+        aks_cluster (str): Name of the AKS cluster
+        configmap_name (str): Name of the ConfigMap to read from
+        format_ (str): Format of the data in the ConfigMap (e.g., "json", "yaml")
+        namespace (str): Kubernetes namespace where the ConfigMap resides (default: "default")
+        prefix_to_add (str): Prefix to add to each key in the ConfigMap
+        content_type (str): Content type to apply to the key-values
+        depth (int): Maximum depth for flattening hierarchical data
+        separator (str): Separator for hierarchical keys
+
+    Returns:
+        list: List of KeyValue objects
+    """
+    try:
+        key_values = []
+        from azure.cli.command_modules.acs.custom import aks_runcommand
+        from azure.cli.command_modules.acs._client_factory import cf_managed_clusters
+
+        # Get the AKS client from the factory
+        cmd.cli_ctx.data['subscription_id'] = aks_cluster["subscription"]
```

Comment:
Does this mean we are allowing the imports from a cluster in a different subscription?

Reply:
Actually, I think my comment can be ignored. 
I was initially under the impression that we only take cluster name as input, but user can also pass the ARM ID , similar to app service. So I guess there need for the restriction. 
Thanks for calling this out @ChristineWanjau !